Diagnostics/FreeBusyChecker/FreeBusyChecker.ps1

```diff
@@ -0,0 +1,3536 @@
+    Write-Host -ForegroundColor White "`n  Parameter: Help"
+    Write-Host  "                 : Use the Help parameter to use display valid parameter Options. `n`n"
+}
+
+if ($Help) {
+    Write-Host $bar
+    ShowHelp
+    $bar
+    exit
+}
+
+Add-PSSnapin microsoft.exchange.management.powershell.Snapin
+Import-Module ActiveDirectory
+
+#Install-Module -Name ExchangeOnlineManagement
+
+# Check if the ExchangeOnlineManagement module is already installed
+if (-not (Get-Module -Name ExchangeOnlineManagement -ListAvailable)) {
+    # If installed Disconnect
+    Disconnect-ExchangeOnline -Confirm:$False
+}
+#DisConnect-ExchangeOnline -Confirm:$False
+
+[System.Console]::Clear()
+$countOrgRelIssues = (0)
+$Script:FedTrust = $null
+$Script:AutoDiscoveryVirtualDirectory = $null
+$Script:OrgRel
+$Script:SPDomainsOnprem
+$AvailabilityAddressSpace = $null
+$Script:WebServicesVirtualDirectory = $null
+$ConsoleWidth = $Host.UI.RawUI.WindowSize.Width
+
+$bar = " =================================================================================================================="
+
+$LogFile = "$PSScriptRoot\FreeBusyChecker.txt"
+$startingDate = (Get-Date -Format yyyyMMdd_HHmmss)
+$LogFileName = [System.IO.Path]::GetFileNameWithoutExtension($LogFile) + "_" + `
+    $startingDate + ([System.IO.Path]::GetExtension($LogFile))
+$htmlFile = "$PSScriptRoot\FBCheckerOutput_$($startingDate).html"
+Write-Host " `n`n "
+Start-Transcript -Path $LogFileName -Append
+Write-Host $bar
+Write-Host -ForegroundColor Green " `n  Free Busy Configuration Information Checker `n "
+Write-Host -ForegroundColor White "   Version 1 `n "
+Write-Host -ForegroundColor Green "  Loading Parameters..... `n "
+#Parameter input
+$UserOnline = Get-RemoteMailbox -ResultSize 1 -WarningAction SilentlyContinue
+$UserOnline = $UserOnline.RemoteRoutingAddress.SmtpAddress
+$ExchangeOnlineDomain = ($UserOnline -split "@")[1]
+
+if ($ExchangeOnlineDomain -like "*.mail.onmicrosoft.com") {
+    $ExchangeOnlineAltDomain = (($ExchangeOnlineDomain.Split(".")))[0] + ".onmicrosoft.com"
+}
+
+else {
+    $ExchangeOnlineAltDomain = (($ExchangeOnlineDomain.Split(".")))[0] + ".mail.onmicrosoft.com"
+}
+# $UserOnPrem = Get-mailbox -ResultSize 1 -WarningAction SilentlyContinue | Where-Object { ($_.EmailAddresses -like "*" + $ExchangeOnlineDomain ) }
+$temp = "*" + $ExchangeOnlineDomain
+$UserOnPrem = ""
+
+$UserOnPrem = Get-mailbox -ResultSize 2 -WarningAction SilentlyContinue -Filter 'EmailAddresses -like $temp -and HiddenFromAddressListsEnabled -eq $false'
+$UserOnPrem = $UserOnPrem[1].PrimarySmtpAddress.Address
+$Script:ExchangeOnPremDomain = ($UserOnPrem -split "@")[1]
+$EWSVirtualDirectory = Get-WebServicesVirtualDirectory -ErrorAction SilentlyContinue
+
+if ($EWSVirtualDirectory.externalURL.AbsoluteUri.Count -gt 1) {
+    $Script:ExchangeOnPremEWS = ($EWSVirtualDirectory.externalURL.AbsoluteUri)[0]
+}
+
+else {
+    $Script:ExchangeOnPremEWS = ($EWSVirtualDirectory.externalURL.AbsoluteUri)
+}
```

Comment:
Maybe this should be another parameter? 

Don't agree with getting all EWS Virtual Directories this way. very long and expensive of a call when you are in a large environment.